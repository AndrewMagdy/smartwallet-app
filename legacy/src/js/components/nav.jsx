import PropTypes from 'prop-types';
import React from 'react'

import Tab from 'material-ui/Tab'
import Tabs from 'material-ui/Tab'

export default class extends React.Component {
  static contextTypes = {
    router: PropTypes.object
  };

  static propTypes = {
    activeTab: PropTypes.any
  };

  _handleTabsChange = (tab) => {
    this.context.router.push(`/${tab}`)
  };

  render() {
    return (
      <Tabs valueLink={{
        value: this.props.activeTab,
        requestChange: this._handleTabsChange
      }} {...this.props}>
        <Tab label="Graph" value="graph" />
      </Tabs>
    )
  }
}
