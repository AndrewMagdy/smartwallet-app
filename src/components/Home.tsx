import * as React from 'react'
import {View, Text} from 'react-native'
import {connect} from 'react-redux'

export interface HomeProps {
  did: string;
  setDid:(did: string) => {};
}

class Home extends React.Component<HomeProps, {}> {
  componentDidMount() {
    setTimeout(this.props.setDid("hello"), 5000)
  }

  render() {
    return (
     <View>
       <Text> {this.props.did} </Text>
     </View>
    )
  }
}

const mapStateToProps = (state:any) => {
  return {
    did: state.did
  }
}
const mapDispatchToProps = (dispatch: any) => {
  return {
    setDid: (did:string) => dispatch({type: 'SET_DID', value: did})
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)