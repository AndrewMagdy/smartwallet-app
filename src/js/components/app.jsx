import React from 'react'
import {Layout, Header, HeaderRow, Content} from 'react-mdl'

import LeftNav from 'components/left-nav/nav.jsx'
import Profile from 'components/accounts/profile.jsx'

import AppNav from 'components/nav.jsx'
import GraphNav from 'components/graph/nav.jsx'
import GraphSearch from 'components/graph/search.jsx'

import ChatNav from 'components/chat/nav.jsx'

import ContactsNav from 'components/contacts/nav.jsx'

import ProjectsNav from 'components/projects/nav.jsx'

let components = {
  '/graph': {
    title: 'Graph',
    nav: <GraphNav/>,
    search: <GraphSearch/>
  },
  '/chat': {
    title: 'Chat',
    nav: <ChatNav/>
  },
  '/contacts': {
    title: 'Contacts',
    nav: <ContactsNav/>
  },
  '/projects': {
    title: 'Projects',
    nav: <ProjectsNav/>
  }
}

let App = React.createClass({

  contextTypes: {
    router: React.PropTypes.func
  },

  getInitialState() {
    return this.getComponent()
  },

  componentDidUpdate(prevProps) {
    if (prevProps.location.pathname !== this.props.location.pathname)
      this.setState(this.getComponent())
  },

  getComponent() {
    return components[this.props.location.pathname]
  },

  render() {
    return (
      <div className="jlc-app">
        <Layout fixedHeader={true} fixedTabs={true}>
          <Header className="jlc-app-header">
            <HeaderRow title={this.state.title}>
              {this.state.nav}
            </HeaderRow>
            <AppNav/>
          </Header>
          {this.state.search}
          <LeftNav/>
          <Content>
            {this.props.children}
          </Content>
        </Layout>
        <Profile/>
      </div>
    )
  }

})

export default App
