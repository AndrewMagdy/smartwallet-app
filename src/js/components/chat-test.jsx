// This component is only meant for test purposes
import React from 'react'
import WebIDAgent from 'lib/agents/webid.js'
import ChatAgent from 'lib/agents/chat.js'
import {Button} from 'react-mdl'

import {linkToState} from 'lib/util'

/*
 - on reload it creates 4 identities
 - you can choose to create a conversation between any of them
 - you can choose to post a message to conversation by any of the participants
*/

let testIdentities = [
    {
      username: 'justas',
      name: 'Justas',
      email: 'justas@gmail.com'
    },
    {
      username: 'eelco',
      name: 'Eelco',
      email: 'eelco@gmail.com'
    },
    {
      username: 'joachim',
      name: 'Joachim',
      email: 'joachim@gmail.com'
    },
    {
      username: 'christian',
      name: 'Christian',
      email: 'christian@gmail.com'
    }
]

let wia = new WebIDAgent()
let chatAgent = new ChatAgent()
let ChatTest = React.createClass({
  getInitialState: function() {
    return {
      conversationUrl: null,
      conversationToLoad: null,
      threads: [],
      messages: [],
      inboxId: null,
      participants: testIdentities[0].username,
      author: testIdentities[0].username,
      message: '',
      testDataPrepared: false
    }
  },
  // just for testing, to refresh dataset
  _recreateUser: function(username, name, email) {
    let webidContainer = `https://localhost:8443/${username}`
    return wia.delete(webidContainer)
      .catch(() => true) //this doesn't work
      .then(() => wia.fakeSignup(username, name, email))
  },

  componentDidMount: function() {
    return Promise.all(testIdentities.map((i) => this._recreateUser(i.username, i.name, i.email)))
      .then(() => {
        console.log('prepared test data.')
        this.setState({
          conversationUrl: this.state.conversationUrl,
          conversationToLoad: this.state.conversationToLoad,
          threads: this.state.threads,
          messages: this.state.messages,
          inboxId: this.state.inboxId,
          participants: this.state.participants,
          author: this.state.author,
          message: this.state.message,
          testDataPrepared: true
        })

      })
  },
  onCreateConversationClick: function() {
    console.log('Creating conversation...')
    console.log(this.state.participants)
    let participants = this.state.participants.split(',').map((username) => `https://localhost:8443/${username}/profile/card#me`)
    let initiator = participants[0]
    chatAgent.createConversation(initiator, participants)
      .then((conversationUrl) => {
        console.log('conversation created')
        console.log(initiator)
        console.log(participants)
        this.setState({
          conversationUrl: conversationUrl,
          conversationToLoad: conversationUrl,
          threads: this.state.threads,
          messages: this.state.messages,
          inboxId: this.state.inboxId,
          participants: this.state.participants,
          author: this.state.author,
          message: this.state.message,
          testDataPrepared: this.state.testDataPrepared
        })
      })

    //TODO: create conversation resource and link it to all participant inboxes
  },

  onPostClick: function() {
    console.log(`Post message ${this.state.message} as ${this.state.author}`)

    let author = `https://localhost:8443/${this.state.author}/profile/card#me`

    return chatAgent.postMessage(this.state.conversationUrl, author, this.state.message)
      .then(() => {
        console.log('posted message')
      })

  },

  onInboxLoadClick: function() {
    console.log(`Loading inbox ${this.state.inboxId} ...`)
    return chatAgent.getInboxConversations(`https://localhost:8443/${this.state.inboxId}/profile/card#me`)
      .then((conversations) => {
        this.setState({
          conversationUrl: this.state.conversationUrl,
          conversationToLoad: this.state.conversationToLoad,
          threads: conversations,
          messages: this.state.messages,
          inboxId: this.state.inboxId,
          participants: this.state.participants,
          author: this.state.author,
          message: this.state.message,
          testDataPrepared: this.state.testDataPrepared
        })
      })
  },

  onLoadThreadClick: function() {
    console.log(`Loading thread ${this.state.conversationToLoad} ...`)
    return chatAgent.getConversationMessages(this.state.conversationToLoad)
      .then((messages) => {
        console.log(messages)
        this.setState({
          conversationUrl: this.state.conversationUrl,
          conversationToLoad: this.state.conversationToLoad,
          threads: this.state.threads,
          messages: messages,
          inboxId: this.state.inboxId,
          participants: this.state.participants,
          author: this.state.author,
          message: this.state.message,
          testDataPrepared: this.state.testDataPrepared
        })
      })
  },

  render: function() {
    let testDataText = this.state.testDataPrepared ? 'Test data prepared.' : 'Preparing test data'
    return (
      <div className="profile">
        <span>Available participants: </span>
        {testIdentities.map(function(i) {
          return (
            <div><b key={i.email}>{i.username}</b><br/></div>
          )
        })}

        <span>Enter comma separated participants:</span>
        <input type="text" onChange={linkToState(this, 'participants')}></input>
        <br/>
        <span>{testDataText}</span>
        <br/>
        <span>{this.state.conversationUrl}</span>
        <br/>
        <Button onClick={this.onCreateConversationClick}>
          Create conversation
        </Button>
        <br/>
        <span>Author</span>
        <br/>
        <input type="text" onChange={linkToState(this, 'author')}></input>
        <br/>
        <span>Message</span>
        <br/>
        <input type="text" onChange={linkToState(this, 'message')}></input>
        <br/>

        <Button onClick={this.onPostClick}>
          Post
        </Button>

        <br/>
        <input type="text" onChange={linkToState(this, 'inboxId')}></input>
        <br/>
        <Button onClick={this.onInboxLoadClick}>
          Load Inbox
        </Button>
        <br/>
        <span>Threads: </span>
        {this.state.threads.map(function(t, index) {
          return (
            <div><b key={index}>{t}</b><br/></div>
          )
        })}
        <br/>

        <br/>
        <input type="text" onChange={linkToState(this, 'conversationToLoad')}></input>
        <br/>
        <span>Messages: </span>
        {this.state.messages.map(function(m, index) {
          return (
            <div><b key={index}>{m.created} {m.author} says: {m.content}</b><br/></div>
          )
        })}
        <br/>
        <Button onClick={this.onLoadThreadClick}>
          Load Thread
        </Button>
      </div>
    )
  }
})

export default ChatTest
