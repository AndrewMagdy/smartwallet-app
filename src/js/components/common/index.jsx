export { default as IconIdCard } from './icon-idcard'
export { default as IconPassport } from './icon-passport'
