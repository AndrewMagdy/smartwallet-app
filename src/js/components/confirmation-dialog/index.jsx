import React from 'react'
import PropTypes from 'prop-types';
import Radium from 'radium'
import { connect } from 'redux_state/utils'

import FlatButton from 'material-ui/FlatButton'
import Dialog from 'material-ui/Dialog'

const STYLES = {
  buttonRigth: {
    textAlign: 'center'
  },
  buttonLeft: {
    textAlign: 'center'
  },
  dialogContainer: {
    textAlign: 'center'
  }
}

@connect({
  props: ['confirm'],
  actions: ['confirmation-dialog:closeConfirmDialog']
})
@Radium
export default class ConfirmationDialog extends React.Component {
  static propTypes = {
    closeConfirmDialog: PropTypes.func,
    confirm: PropTypes.object,
    cancelActionText: PropTypes.string
  }

  _handleConfirmAction() {
    this.props.closeConfirmDialog()
    console.log(this.props.confirm)
    if (this.props.confirm.callback)
      this.props.confirm.callback() // Action when the user confirms
  }

  _handleConfirmCancel() {
    this.props.closeConfirmDialog()
  }

  render() {
    const confirmHandler = () => this._handleConfirmAction()
    const cancelHandler = () => this._handleConfirmCancel()

    const confirmActions = [
      <FlatButton
        label={this.props.confirm.cancelActionText || 'Cancel'}
        secondary
        onTouchTap={cancelHandler}
        style={STYLES.buttonRigth}
      />,
      <FlatButton
        label={this.props.confirm.primaryActionText}
        secondary
        onTouchTap={confirmHandler}
        style={STYLES.buttonLeft}
      />
    ]

    return <Dialog
      actions={confirmActions}
      title={this.props.confirm.title}
      modal={false}
      open={this.props.confirm.open}
      onRequestClose={this.handleClose}
      contentStyle={this.props.confirm.style}
      actionsContainerStyle={STYLES.dialogContainer}
    >
      {this.props.confirm.message}
    </Dialog>
  }
}
