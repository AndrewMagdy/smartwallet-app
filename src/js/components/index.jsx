import React from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'

import RaisedButton from 'material-ui/RaisedButton'
import IconButton from 'material-ui/IconButton'

import Carousel from 'components/common/carousel.jsx'
import IndicatorDots from 'components/common/indicator-dots.jsx'
import {connect} from 'redux_state/utils'

import {routes} from 'routes'
import {theme} from 'styles'
import { Container, Header, SideNote, InfoLink } from './structure'

const message = (<div>
  <div>A Blockchain is a distributed database which
  Integrity (security from manipulation) is given by a timestamp and
  a link to the previous data set. These days it is used for cryptocurrency,
  which means that it is as secure as your bank account
  </div><br />
  <div>We use this technology to lock your data. So only you can have
  access to it.</div></div>)

const dialogBlockchain = {
  primaryActionText: 'MORE INFO',
  cancelActionText: 'ALL RIGHT',
  message: message,
  style: {
    actionsContainerStyle: {
      textAlign: 'center'
    }
  },
  callback: () => {
    window.open('https://en.wikipedia.org/wiki/Blockchain', '_blank')
  },
  title: 'What is a blockchain?'
}

class Index extends React.Component {
  static contextTypes = {
    router: PropTypes.object,
    username: PropTypes.string,
    muiTheme: PropTypes.object
  };

  static propTypes = {
    openConfirmDialog: PropTypes.func
  };

  getStyles = () => {
    let {muiTheme} = this.context

    let styles = {
      logoImg: {
        flex: 1,
        width: '150px',
        userSelect: 'none',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'contain',
        '@media screen and (min-width: 1200px)': {
          width: '250px'
        }
      },
      intro: {
        flex: 1,
        textAlign: 'center'
      },
      slide: {
        backgroundColor: muiTheme.jolocom.gray4,
        padding: '24px 24px 48px',
        height: '95%',
        boxSizing: 'border-box',
        display: 'flex',
        flexDirection: 'column',
        userSelect: 'none',
        alignItems: 'center'
      },
      onboardImg: {
        flex: 1,
        width: '300px',
        marginBottom: '0',
        userSelect: 'none',
        backgroundPosition: 'center bottom',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'contain',
        '@media screen and (min-width: 1200px)': {
          width: '600px'
        }
      },
      title: {
        color: theme.textStyles.headline.color,
        fontSize: theme.textStyles.headline.fontSize,
        fontWeight: theme.textStyles.headline.fontWeight,
        marginBottom: '10px',
        maxWidth: '100%',
        '@media screen and (min-width: 1200px)': {
          fontSize: '24pt'
        }
      },
      subtitle: {
        color: theme.textStyles.textCopy.color,
        fontSize: theme.textStyles.textCopy.fontSize,
        fontWeight: theme.textStyles.textCopy.fontWeight,
        '@media screen and (min-width: 1200px)': {
          fontSize: '14pt'
        }
      },
      actions: {
        display: 'flex',
        alignItems: 'stretch',
        padding: '16px',
        maxWidth: '80%',
        width: '440px',
        margin: '0 auto 0px'
      },
      signup: {
        margin: '10px',
        width: '200px'
      },
      login: {
        margin: '10px',
        width: '200px'
      },
      previous: {
        position: 'absolute',
        top: '50%',
        left: '20px',
        marginTop: '-12px',
        zIndex: 100,
        '@media screen and (max-width: 468px)': {
          display: 'none'
        }

      },
      next: {
        position: 'absolute',
        top: '50%',
        right: '20px',
        marginTop: '-12px',
        zIndex: 100,
        '@media screen and (max-width: 468px)': {
          display: 'none'
        }
      },
      embeddedLink: {
        color: theme.palette.accent1Color,
        minWidth: '0px',
        paddingLeft: '5px',
        paddingRight: '5px'
      }
    }

    return styles
  };

  render() {
    let styles = this.getStyles()
    return (
      <Container>
        <div style={styles.previous}>
          <IconButton
            iconClassName="material-icons"
            onClick={this._handlePrevious}
          >chevron_left</IconButton>
        </div>

        <div style={styles.next}>
          <IconButton
            iconClassName="material-icons"
            onClick={this._handleNext}
          >chevron_right</IconButton>
        </div>

        <Carousel
          ref={this._setCarouselRef}
          auto={false}
          style={styles.intro} indicator={IndicatorDots}>
          <div style={styles.slide}>
            <div style={Object.assign({}, styles.logoImg, {
              backgroundImage: 'url(img/logo_start.svg)'
            })} />
          </div>
          <div style={styles.slide}>
            <div style={Object.assign({}, styles.onboardImg, {
              backgroundImage: 'url(img/img_onboarding-01.svg)'
            })} />
            <Header
              title="Create an independent and secure digital identity." />
            <SideNote>
              Collect your data at a secure place.
              Itâ€™s yours, so only you own it!
            </SideNote>
          </div>
          <div style={styles.slide}>
            <div style={Object.assign({}, styles.onboardImg, {
              backgroundImage: 'url(img/img_onboarding-02.svg)'
            })} />
            <Header title="Have all your data at your fingertips." />
            <SideNote>
              See all your data in one safe place. Pull the
              plug and your data is only yours.
            </SideNote>
          </div>
          <div style={styles.slide}>
            <div style={Object.assign({}, styles.onboardImg, {
              backgroundImage: 'url(img/img_onboarding-03.svg)'
            })} />
            <Header title="Be aware of the information you share." />
            <SideNote>
              See what you shared with whom.
              Have total control over your data.
            </SideNote>
          </div>
          <div style={styles.slide}>
            <div style={Object.assign({}, styles.onboardImg, {
              backgroundImage: 'url(img/img_onboarding-04.svg)'
            })} />
            <Header
              title={'Our Wallet keeps your data as safe' +
              ' as your bank account.'} />
            <SideNote>
              We use the latest encryption technology and
              <span style={styles.embeddedLink}
                onClick={() => {
                  this.props.openConfirmDialog(dialogBlockchain)
                }}>
              blockchain</span>
              to store your sensitive data.
            </SideNote>
          </div>
          <div style={styles.slide}>
            <div style={Object.assign({}, styles.onboardImg, {
              backgroundImage: 'url(img/img_onboarding-05.svg)'
            })} />
            <Header
              style={{marginTop: '0'}}
              title={'Security is hard to ' +
                'maintain, that is why the storage costs.'} />
            <SideNote>
              The storage of your data is payed
              in ether, a webbased currency. But only the
              change of data costs.
            </SideNote>
          </div>
        </Carousel>

        <div style={styles.actions}>
          <RaisedButton
            secondary
            label="Sign up"
            style={styles.signup}
            onClick={this._handleSignup}
          />
          <RaisedButton
            label="Log in"
            style={styles.login}
            onClick={this._handleLogin}
          />
        </div>
        <InfoLink
          info="With signing up you agree with our"
          link="AGB"
          to="" />
      </Container>
    )
  }

  _setCarouselRef = (c) => {
    this.carousel = c
  };

  _handlePrevious = () => {
    this.carousel.slideTowards('right')
  };

  _handleNext = () => {
    this.carousel.slideTowards('left')
  };

  _handleSignup = () => {
    this.context.router.push(routes.signup)
  };

  _handleLogin = () => {
    this.context.router.push('/login')
  };
}

export default connect({
  actions: [
    'confirmation-dialog:openConfirmDialog',
    'confirmation-dialog:closeConfirmDialog'
  ]
})(Radium(Index))
