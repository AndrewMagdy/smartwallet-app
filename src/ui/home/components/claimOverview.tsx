import React from 'react'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import { StyleSheet, TouchableOpacity, Text, View, ScrollView } from 'react-native'
import { Container, Block } from 'src/ui/structure'
import { ClaimCard } from 'src/ui/home/components/claimCard'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { ReactNode } from 'react'
import { ClaimsState } from 'src/reducers/account'
import { Claim } from 'src/actions/account/helper'

interface Props {
  claims: ClaimsState
  scanning: boolean
  onScannerStart: () => void
  openClaimDetails: (id: string) => void
}

interface State {
}

const categoryDisplayMap: {[key: string] : string} = {
  personal: 'Personal / general',
  contact: 'Contact',
  other: 'Miscellaneous'
}

const styles = StyleSheet.create({
  icon: {
    margin: "20%"
  },
  iconContainer: {
    height: 55,
    width: 55,
    borderRadius: 35,
    backgroundColor: JolocomTheme.primaryColorPurple,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 8,
  },
  actionButtonContainer: {
    position: 'absolute',
    right: '3%',
    bottom: '5%',
    alignItems: 'flex-end'
  },
  sectionHeader: {
    fontSize: 17,
    textAlign: 'left'
  },
  componentContainer: {
    flex: 1,
    padding: 0
  },
  scrollComponent: {
    width: '100%'
  },
  sectionContainer: {
    marginBottom: 8,
    marginTop: 27,
    marginLeft: 16,
    marginRight: 16
  }
})

export class ClaimOverview extends React.Component<Props, State> {

  renderClaimCards = (category: string) : ReactNode => {
    const { openClaimDetails, claims } = this.props
    const categoryClaims = claims.savedClaims[category]
    return categoryClaims.map((claim: Claim) => {
      switch(claim.multiLine) {
      case (true):
        return // TODO: handle multiLine claims e.g. address for later
      default:
        return (
          <ClaimCard
            key={ claim.id }
            openClaimDetails={ openClaimDetails }
            claimItem={ claim }
          />
        )
      }
    })
  }

  render() {
    const { claims } = this.props
    const claimsCategories = Object.keys(claims.savedClaims)
    const content = claims.loading ? ( <View><Text>Loading</Text></View>) :
      (claimsCategories.map((category: string) => {
          return (
            <View key={category}>
              <View style={ styles.sectionContainer }>
                <Text style={ styles.sectionHeader }>{ categoryDisplayMap[category] }</Text>
              </View>
              { this.renderClaimCards(category) }
            </View>
          )
        })
      )
    
    return (
      <Container style={ styles.componentContainer }>
        <ScrollView style={ styles.scrollComponent }>
          { content }
        </ScrollView>
        <Block style={ styles.actionButtonContainer }>
          <TouchableOpacity
            style={ styles.iconContainer }
            onPress={ this.props.onScannerStart }>
            <Icon
              style={ styles.icon }
              size={ 30 }
              name='qrcode-scan'
              color="white"
            />
          </TouchableOpacity>
        </Block>
      </Container>
    )
  }
}
