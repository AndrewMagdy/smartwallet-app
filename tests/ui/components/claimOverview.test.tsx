import React from 'react'
import { ClaimOverview } from 'src/ui/home/components/claimOverview'
import { shallow } from 'enzyme'

describe.only('ClaimsOverview component', ()=> {
  const COMMON_PROPS = {
    claims: {
      loading: false,
      decoratedCredentials: {
        'Other': [{
            displayName: 'Age',
            type: ['Credential', 'ProofOfAge'],
            claims: [{
              id: 'default1',
              name: 'Age',
              value: '50',
            }],
        }],
        'Personal' : [{
            displayName: 'Name',
            type: ['Credential', 'ProofOfNameCredential'],
            claims: [{
              id: 'default1',
              name: 'name',
              value: 'name',
            }],
          }],
        'Contact': [{
            displayName: 'E-mail',
            type: ['Credential', 'ProofOfEmailCredential'],
            claims: [{
              id: 'default2',
              name: 'email',
              value: ''
            }],
          },
          {
            displayName: 'Phone',
            type: ['Credential', 'ProofOfMobilePhoneNumberCredential'],
            claims: [{
              id: 'default3',
              name: 'phone',
              value: ''
            }],
          }]
      }
    },
    scanning: false,
    onScannerStart: () => null,
    openClaimsDetails: () => null,
  }
  it('matches the snapshot on render', () => {

    const rendered = shallow(<ClaimOverview {...COMMON_PROPS}/>)
    expect(rendered).toMatchSnapshot()
  })
})
