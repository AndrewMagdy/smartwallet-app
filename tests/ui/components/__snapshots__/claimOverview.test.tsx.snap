// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimsOverview component matches the snapshot on render 1`] = `
<Container
  style={
    Object {
      "flex": 1,
      "padding": 0,
    }
  }
>
  <ScrollView
    contentContainerStyle={Object {}}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Component
      key="Contact"
    >
      <Component
        style={
          Object {
            "marginBottom": 8,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 27,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "TT Commons",
              "fontSize": 17,
              "textAlign": "left",
            }
          }
        >
          Contact
        </Text>
      </Component>
      <ClaimCard
        claimItem={
          Object {
            "claims": Array [
              Object {
                "id": "default2",
                "name": "email",
                "value": "",
              },
            ],
            "displayName": "E-mail",
            "type": Array [
              "Credential",
              "ProofOfEmailCredential",
            ],
          }
        }
      />
      <ClaimCard
        claimItem={
          Object {
            "claims": Array [
              Object {
                "id": "default3",
                "name": "phone",
                "value": "",
              },
            ],
            "displayName": "Phone",
            "type": Array [
              "Credential",
              "ProofOfMobilePhoneNumberCredential",
            ],
          }
        }
      />
    </Component>
    <Component
      key="Personal"
    >
      <Component
        style={
          Object {
            "marginBottom": 8,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 27,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "TT Commons",
              "fontSize": 17,
              "textAlign": "left",
            }
          }
        >
          Personal
        </Text>
      </Component>
      <ClaimCard
        claimItem={
          Object {
            "claims": Array [
              Object {
                "id": "default1",
                "name": "name",
                "value": "name",
              },
            ],
            "displayName": "Name",
            "type": Array [
              "Credential",
              "ProofOfNameCredential",
            ],
          }
        }
      />
    </Component>
    <Component
      key="Other"
    >
      <Component
        style={
          Object {
            "marginBottom": 8,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 27,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "TT Commons",
              "fontSize": 17,
              "textAlign": "left",
            }
          }
        >
          Other
        </Text>
      </Component>
      <ClaimCard
        claimItem={
          Object {
            "claims": Array [
              Object {
                "id": "default1",
                "name": "Age",
                "value": "50",
              },
            ],
            "displayName": "Age",
            "type": Array [
              "Credential",
              "ProofOfAge",
            ],
          }
        }
      />
    </Component>
  </ScrollView>
  <Block
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": "5%",
        "position": "absolute",
        "right": "3%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#942f51",
          "borderRadius": 35,
          "elevation": 8,
          "height": 55,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 55,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="qrcode-scan"
        size={30}
        style={
          Object {
            "margin": "20%",
          }
        }
      />
    </TouchableOpacity>
  </Block>
</Container>
`;
